source/modulo_components/modulo_components/component_interface.py

```diff
@@ -278,6 +284,86 @@ def set_predicate(self, predicate_name: str, predicate_value: Union[bool, Callab
             return
         self._predicates[predicate_name] = predicate_value
 
+    def add_output(self, signal_name: str, data, message_type: MsgT, fixed_topic=False, default_topic=""):
+        try:
+            parsed_signal_name = parse_signal_name(signal_name)
+            if not parsed_signal_name:
+                raise AddSignalError("Failed to add output: Provide a non empty string as a name.")
+            if parsed_signal_name in self._outputs.keys():
+                raise AddSignalError(f"Output with name '{parsed_signal_name}' already exists")
+            topic_name = default_topic if default_topic else "~/" + parsed_signal_name
+            self.add_parameter(sr.Parameter(parsed_signal_name + "_topic", topic_name, sr.ParameterType.STRING),
+                               f"Output topic name of signal '{parsed_signal_name}'", fixed_topic)
+            translator = None
+            if message_type == Bool or message_type == Float64 or \
+                    message_type == Float64MultiArray or message_type == Int32 or message_type == String:
+                translator = modulo_writers.write_std_msg
+            elif message_type == EncodedState:
+                # TODO switch case to find message type
```

Comment:
This TODO is actually quite important and something we need to discuss. I can see these two options, there are probably others:
- actually implement the switch case
- just add another optional argument that would be used here

The drawback of the second option is that we put the responsibility on the user to provide the correct clproto message type. The drawback of the first option is that it will result in a lot of nested `isinstance` conditions, where I'm not sure if its very robust.